### PR TITLE
Allow tests to explicitly skip backend for unsupported op categories

### PR DESCRIPTION
**Pull request by notoraptor**
Hi @breuleux @abergeron this is a PR to allow to skip tests for backends that don't support some specified op categories.

- I added a file to define op categories (named `computations.py`) into `myia.compile.backends`
- I updated Backend base class and derived class in each backend with a method to check if an op category is supported
- I added a testing decorator `bt` (for `backend testing`), which can receives a list of required op categories and will explicitly skip test for if backend does not support all required categories.
- I used `bt` in `test_universal.py` as example.

What do you think ?

### CODE REVIEW

**Comment by notoraptor** [comment]
@abergeron @breuleux PR updated ! Though Jenkins does not start ...

File: myia/compile/backends/prim_groups.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        P.scalar_tanh,
+        P.scalar_trunc,
+        P.scalar_uadd,
+        P.scalar_usub,
+    ],
+)
+
+
+# Ignored primitives:
+# P.make_handle
```
Comment (abergeron) [inline myia/compile/backends/prim_groups.py:137]: This should be in the universe group.

File: myia/api.py [hunk 2]
```diff
@@ -48,11 +48,6 @@ def __init__(
         pipeline=standard_pipeline,
     ):
         """Initialize a MyiaFunction."""
-        # Change this once relay becomes the default backend.
-        if use_universe and backend not in ("python", "relay"):
-            raise RuntimeError(  # pragma: no cover
-                "Universe is only supported for the relay backend."
-            )
```
Comment (abergeron) [inline myia/api.py:55]: This check should probably stay as it is not for the tests but for end users.  Unless I'm missing something in this PR.

**Review by abergeron — APPROVED** [review]
I'm overall ok, with two minor nits

**Comment by abergeron** [comment]
We still have to make sure that Jenkins runs.

**Comment by notoraptor** [comment]
@abergeron Updated (see latest commit). Jenkins launched !

**Comment by notoraptor** [comment]
Hi @abergeron tests succeed on Jenkins, but coverage fail: https://jenkins.server.mila.quebec/blue/organizations/jenkins/Mila%2Fmyia/detail/PR-383/3/pipeline

Failing files should be ignored based on `codecov.yml` file, but they appear on report: https://github.com/mila-iqia/myia/blob/master/codecov.yml

**Comment by notoraptor** [comment]
Oh, ultimately, all tests passed (even coverage) !

File: myia_backend_relay/myia_backend_relay/relay.py [hunk 3]
```diff
@@ -1071,6 +1071,13 @@ def compile(self, graph, argspec, outspec):
             graph, self.context, self.target, self.exec_kind
         )
 
+    def supports_prim_group(self, prim_group):
+        if prim_group is G.universe_operations:
+            # Relay does not explicitly implement primitive make_handle
+            # but still supports universe operations.
+            return True
```
Comment (breuleux) [inline myia_backend_relay/myia_backend_relay/relay.py:1078]: That's a brittle check, we should accept any primitive group that contains `make_handle`. A better hardcoded fix would be to write `MAP.get(prim) or prim is P.make_handle` as the condition in the line below. Alternatively, set `P.make_handle: True` in `SIMPLE_MAP`, or have an additional set of supported primitives for Relay.

Comment (notoraptor) [reply myia_backend_relay/myia_backend_relay/relay.py]: Updated @breuleux with `MAP.get(prim) or prim is P.make_handle` !

**Pull request merged**